2024-01-28 13:24:27,611: root: INFO: MLTrader : Executing the initialize lifecycle method
2024-01-28 13:24:28,022: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:28,061: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:28,911: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:28,958: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:29,178: root: INFO: Current backtesting datetime 2023-12-15 08:30:00-05:00
2024-01-28 13:24:29,180: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-01-28 13:24:29,181: root: INFO: Current backtesting datetime 2023-12-15 09:30:00-05:00
2024-01-28 13:24:29,181: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-01-28 13:24:29,182: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:29[0m
2024-01-28 13:24:29,182: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:29,192: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:29,578: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:29,592: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:30,093: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:30,106: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:30,688: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:30,739: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:31,362: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:31,374: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:31,878: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:31,890: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:31,897: root: INFO: [32mNew market order of | 106.0 SPY buy | at price $563.3879882812499 of class bracket with status unprocessed was submitted.[0m
2024-01-28 13:24:31,902: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,902: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,902: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,904: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,904: root: INFO: market order of | 106.0 SPY buy | at price $563.3879882812499 of class bracket with status new was sent to broker backtesting
2024-01-28 13:24:31,904: root: INFO: market order of | 106.0 SPY buy | at price $563.3879882812499 of class bracket with status new was sent to broker backtesting
2024-01-28 13:24:31,904: root: INFO: [32mFilled Transaction: buy 106.0 of SPY at 469.48999023 USD per share[0m
2024-01-28 13:24:31,904: root: INFO: market order of | 106.0 SPY buy | at price $563.3879882812499 of class bracket with status new was filled
2024-01-28 13:24:31,909: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,911: root: INFO: Current backtesting datetime 2023-12-15 16:00:00-05:00
2024-01-28 13:24:31,911: root: INFO: Current backtesting datetime 2023-12-16 09:30:00-05:00
2024-01-28 13:24:31,913: root: INFO: [34mMLTrader : The market is not currently open, skipping this trading iteration[0m
2024-01-28 13:24:31,913: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,913: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,914: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,914: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,915: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,916: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,917: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,917: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,918: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,918: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,918: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,919: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,920: root: INFO: Current backtesting datetime 2023-12-16 09:29:00-05:00
2024-01-28 13:24:31,921: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,921: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,922: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,922: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,922: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,923: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,924: root: INFO: Current backtesting datetime 2023-12-16 09:29:00-05:00
2024-01-28 13:24:31,924: root: INFO: MLTrader : Executing the after_market_closes lifecycle method
2024-01-28 13:24:31,924: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,924: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,925: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,925: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,925: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,926: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,927: root: INFO: Current backtesting datetime 2023-12-18 08:30:00-05:00
2024-01-28 13:24:31,928: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,929: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,929: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-01-28 13:24:31,930: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,930: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,930: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,931: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,931: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,932: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,933: root: INFO: Current backtesting datetime 2023-12-18 09:30:00-05:00
2024-01-28 13:24:31,934: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-01-28 13:24:31,936: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,936: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,937: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,938: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,938: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,938: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,938: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,939: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,939: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,940: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,941: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,941: root: INFO: Current backtesting datetime 2023-12-18 16:00:00-05:00
2024-01-28 13:24:31,942: root: INFO: Current backtesting datetime 2023-12-19 09:30:00-05:00
2024-01-28 13:24:31,943: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,943: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,944: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,944: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,944: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,945: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,945: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,945: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,945: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,946: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,948: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,949: root: INFO: Current backtesting datetime 2023-12-19 16:00:00-05:00
2024-01-28 13:24:31,949: root: INFO: Current backtesting datetime 2023-12-20 09:30:00-05:00
2024-01-28 13:24:31,950: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,951: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,951: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,952: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,952: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,952: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,953: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,953: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,953: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,954: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,955: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,956: root: INFO: Current backtesting datetime 2023-12-20 16:00:00-05:00
2024-01-28 13:24:31,956: root: INFO: Current backtesting datetime 2023-12-21 09:30:00-05:00
2024-01-28 13:24:31,957: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,957: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,958: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,959: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,959: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,959: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,959: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,960: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,960: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,960: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,962: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,963: root: INFO: Current backtesting datetime 2023-12-21 16:00:00-05:00
2024-01-28 13:24:31,963: root: INFO: Current backtesting datetime 2023-12-22 09:30:00-05:00
2024-01-28 13:24:31,965: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,965: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,966: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,966: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,966: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,966: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,967: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,967: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,967: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,968: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,969: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,970: root: INFO: Current backtesting datetime 2023-12-22 16:00:00-05:00
2024-01-28 13:24:31,970: root: INFO: Current backtesting datetime 2023-12-23 09:30:00-05:00
2024-01-28 13:24:31,972: root: INFO: [34mMLTrader : The market is not currently open, skipping this trading iteration[0m
2024-01-28 13:24:31,972: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,972: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,973: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,973: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,973: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,973: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,974: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,975: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,975: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,975: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,975: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,976: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,977: root: INFO: Current backtesting datetime 2023-12-23 09:29:00-05:00
2024-01-28 13:24:31,978: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,978: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,979: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,979: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,979: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,980: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,981: root: INFO: Current backtesting datetime 2023-12-23 09:29:00-05:00
2024-01-28 13:24:31,981: root: INFO: MLTrader : Executing the after_market_closes lifecycle method
2024-01-28 13:24:31,981: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,981: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,982: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,982: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,982: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,983: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,984: root: INFO: Current backtesting datetime 2023-12-26 08:30:00-05:00
2024-01-28 13:24:31,985: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,985: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,986: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-01-28 13:24:31,986: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,986: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,987: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,987: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,987: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,988: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,989: root: INFO: Current backtesting datetime 2023-12-26 09:30:00-05:00
2024-01-28 13:24:31,989: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-01-28 13:24:31,991: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,992: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,992: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,993: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,993: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,993: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,994: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,994: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:31,994: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,994: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:31,996: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:31,997: root: INFO: Current backtesting datetime 2023-12-26 16:00:00-05:00
2024-01-28 13:24:31,997: root: INFO: Current backtesting datetime 2023-12-27 09:30:00-05:00
2024-01-28 13:24:31,999: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:31[0m
2024-01-28 13:24:31,999: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,000: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,000: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:32[0m
2024-01-28 13:24:32,000: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,000: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,001: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,001: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,001: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,002: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,003: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:32,004: root: INFO: Current backtesting datetime 2023-12-27 16:00:00-05:00
2024-01-28 13:24:32,004: root: INFO: Current backtesting datetime 2023-12-28 09:30:00-05:00
2024-01-28 13:24:32,005: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:32[0m
2024-01-28 13:24:32,005: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,006: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,007: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:32[0m
2024-01-28 13:24:32,007: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,007: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,007: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,008: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,008: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,008: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,009: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:32,010: root: INFO: Current backtesting datetime 2023-12-28 16:00:00-05:00
2024-01-28 13:24:32,010: root: INFO: Current backtesting datetime 2023-12-29 09:30:00-05:00
2024-01-28 13:24:32,012: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 13:24:32[0m
2024-01-28 13:24:32,012: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,013: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,013: root: INFO: [34mMLTrader : Trading iteration ended at 2024-01-28 13:24:32[0m
2024-01-28 13:24:32,014: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,014: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,014: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,014: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,015: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,015: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,016: root: INFO: MLTrader : [34mSleeping for 86400 seconds[0m
2024-01-28 13:24:32,017: root: INFO: Current backtesting datetime 2023-12-29 16:00:00-05:00
2024-01-28 13:24:32,017: root: INFO: Current backtesting datetime 2023-12-30 09:30:00-05:00
2024-01-28 13:24:32,019: root: INFO: [34mMLTrader : The market is not currently open, skipping this trading iteration[0m
2024-01-28 13:24:32,019: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,019: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,019: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,020: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,020: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,020: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,021: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,021: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,022: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,022: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,022: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,023: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,024: root: INFO: Current backtesting datetime 2023-12-30 09:29:00-05:00
2024-01-28 13:24:32,024: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,024: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,025: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,025: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,025: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,026: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,027: root: INFO: Current backtesting datetime 2023-12-30 09:29:00-05:00
2024-01-28 13:24:32,027: root: INFO: MLTrader : Executing the after_market_closes lifecycle method
2024-01-28 13:24:32,027: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,027: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,028: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,028: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,028: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,029: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,030: root: INFO: Current backtesting datetime 2024-01-02 08:30:00-05:00
2024-01-28 13:24:32,030: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,031: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,032: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-01-28 13:24:32,032: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,032: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,033: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,033: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,033: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,034: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,035: root: INFO: Current backtesting datetime 2024-01-02 09:30:00-05:00
2024-01-28 13:24:32,035: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-01-28 13:24:32,036: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,036: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,036: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,037: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,037: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,037: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,038: root: INFO: Current backtesting datetime 2024-01-02 15:59:00-05:00
2024-01-28 13:24:32,039: root: INFO: MLTrader : Executing the before_market_closes lifecycle method
2024-01-28 13:24:32,039: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,039: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,039: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,040: root: INFO: Getting historical prices for SPY, 1 bars, 
2024-01-28 13:24:32,040: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,040: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 13:24:32,041: root: INFO: Current backtesting datetime 2024-01-02 16:00:00-05:00
2024-01-28 13:24:32,041: root: INFO: MLTrader : Executing the after_market_closes lifecycle method
2024-01-28 13:24:32,042: root: INFO: MLTrader : Executing the on_strategy_end lifecycle method
2024-01-28 13:24:32,414: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:32,427: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:32,717: root: INFO: Backtesting finished
2024-01-28 13:24:33,954: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:33,966: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:34,536: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:34,549: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:35,151: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:35,164: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:35,665: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:35,678: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:36,278: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:36,292: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:36,892: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:36,906: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:37,507: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:37,520: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:38,122: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:38,137: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:38,670: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:38,685: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:39,257: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:39,272: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:39,779: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:39,791: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:40,377: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:40,390: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:40,921: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:40,934: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:41,535: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 170, in _run_forever
    await self._start_ws()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 130, in _start_ws
    await self._connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/alpaca/trading/stream.py", line 61, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1112, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1145, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 574, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 556, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)
2024-01-28 13:24:41,561: alpaca.trading.stream: INFO: starting trading websocket connection
2024-01-28 13:24:41,841: backtest_stats: INFO: Backtest took 0:00:14.230876 for a speed of 0.000
